newdle/client/src/components/answer/AnswerPage.js

```diff
@@ -1,15 +1,41 @@
-import {Grid, Checkbox, Header} from 'semantic-ui-react';
-import React from 'react';
+import {Grid, Checkbox, Input, Button, Segment, Message} from 'semantic-ui-react';
+import PropTypes from 'prop-types';
+import React, {useState} from 'react';
+import {useHistory} from 'react-router';
 import MonthCalendar from './MonthCalendar';
 import Calendar from './Calendar';
-import {getNumberOfTimeslots, getNumberOfAvailableAnswers, getNewdle} from '../../answerSelectors';
+import {
+  getAnswers,
+  getNumberOfTimeslots,
+  getNumberOfAvailableAnswers,
+  getNewdle,
+} from '../../answerSelectors';
 import {useSelector} from 'react-redux';
 import styles from './answer.module.scss';
+import client from '../../client';
+import {useClient} from '../../util/hooks';
 
-export default function AnswerPage() {
+export default function AnswerPage({match: {params}}) {
+  const participantCode = params.partcode;
   const newdle = useSelector(getNewdle);
   const numberOfTimeslots = useSelector(getNumberOfTimeslots);
   const numberOfAvailable = useSelector(getNumberOfAvailableAnswers);
+  const availabilityData = useSelector(getAnswers);
+  const [name, setName] = useState('');
+  const history = useHistory();
+  const [saved, setSaved] = useState(false);
+
+  const [_answerNewdle, submitting, error] = useClient(
+    participantCode ? client.answerNewdleParticipant : client.answerNewdleAnonymous
+  );
+
+  const canSubmit = (participantCode || name.length >= 2) && !submitting && numberOfAvailable > 0;
```

Comment:
Why can't you submit with zero available? I think it makes perfect sense to explicitly state that you are not available at any time.

Reply:
Indeed.